portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -0,0 +1,130 @@
+            this.loggedIn = logstat;
+            this.email = userEmail;
+            this.username = userName;
+        }
+        */
+       
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        //set response to json
+        response.setContentType("application/json;");
+
+        UserService userService = UserServiceFactory.getUserService();
+        Login userlogstat = new Login();
+        if (userService.isUserLoggedIn()) {
+            String email = userService.getCurrentUser().getEmail();
+            String username = getUsername(userService.getCurrentUser().getUserId());
+            String logoutUrl = userService.createLogoutURL("/");
+            userlogstat.setLoggedIn(email, username, logoutUrl);
+        }else{
+            String logoutUrl = userService.createLoginURL("/");
+            userlogstat.setLogoutURL(logoutUrl);
+        }
+        //convert login status to json
+        String json = new Gson().toJson(userlogstat);
+        response.getWriter().println(json);
+
+        /*
+        response.setContentType("text/html");
+        UserService userService = UserServiceFactory.getUserService();//user login api
+        if (userService.isUserLoggedIn()) {
+            String username = getUsername(userService.getCurrentUser().getUserId());
+            out.println("<h1>User Logged-in</h1>");
+            out.println("<p>Username: " + username + "</p>");
+            out.println("<p>Email: " + userService.getCurrentUser().getEmail()+ "</p>");
+            //logout url when logged in
+            String logoutUrl = userService.createLogoutURL("/loginstat");
+            out.println("<p>Logout <a href=\"" + logoutUrl + "\">here</a>.</p>");
+  
+        }else {
+            out.println("<h1>User Not Logged-in</h1>");
+            //logout url when logged in
+            String loginUrl = userService.createLoginURL("/loginstat");
+            out.println("<p>Login <a href=\"" + loginUrl + "\">here</a>.</p>");
+        }
+        */
+        
+    }
+    /*
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        UserService userService = UserServiceFactory.getUserService();
+        if (!userService.isUserLoggedIn()) {
+            response.sendRedirect("/loginstat");
+            return;
+        }
+        String username = request.getParameter("nickname");
+        String id = userService.getCurrentUser().getUserId();
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        Entity entity = new Entity("UserInfo", id);
+        entity.setProperty("id", id);
+        entity.setProperty("nickname", nickname);
+        // The put() function automatically inserts new data or updates existing data based on ID
+        datastore.put(entity);
+
+        response.sendRedirect("/home");
+    }*/
+
+    //get username
```

Comment:
Avoid superfluous comments such as these – if the name of the function makes it trivially obvious as to what goes on, then feel free to leave out any comments.

In this case, a comment can still help convey something helpful such as:

```
// Returns username if the id belongs to a valid user, otherwise returns an empty string.
```